### ISSUE TITLE
Changing anchors

### ISSUE THREAD

**Issue by bluetooth12** [fizyr/keras-retinanet#957]
I am trying to modify the anchor boxes like so: 
in utils\anchors.py 
AnchorParameters.default = AnchorParameters(
    sizes   = [32, 64, 128, 256, 512],
    strides = [8, 16, 32, 64, 128],
    #ratios  = np.array([0.5, 1, 2], keras.backend.floatx()),
    #scales  = np.array([2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)], keras.backend.floatx()),
    ratios  = np.array([0.33, 0.5, 1, 2, 3], keras.backend.floatx()),
    scales  = np.array([0.33, 0.66, 1, 1.2, 1.6], keras.backend.floatx()),
)
It trained. I then generated an inference .h5 file by using 
python keras_retinanet/bin/convert_model.py resnet50_coco_50_adj_weights.h5 resnet50_coco_adj_weights_inference.h5

However, I can't evaluate the inference model. If I try
python keras_retinanet\bin\evaluate.py coco cocoapi resnet50_coco_adj_weights_inference.h5 
I get
![image](https://user-images.githubusercontent.com/8892431/55022497-e6b8f000-4fb7-11e9-85f3-2f2bad48bd4f.png)

I tried looking online but I am using the latest version of keras-retinanet and the suggested changes seem to be already made.

**Comment by bluetooth12** [comment]
In layers\_misc.py 
class Anchors(keras.layers.Layer):
    """ Keras layer for generating achors for a given shape.
    """

    def __init__(self, size, stride, ratios=None, scales=None, *args, **kwargs):
        """ Initializer for an Anchors layer.

        Args
            size: The base size of the anchors to generate.
            stride: The stride of the anchors to generate.
            ratios: The ratios of the anchors to generate (defaults to AnchorParameters.default.ratios).
            scales: The scales of the anchors to generate (defaults to AnchorParameters.default.scales).
        """
        self.size   = size
        self.stride = stride
        self.ratios = ratios
        self.scales = scales

        if ratios is None:
            self.ratios  = utils_anchors.AnchorParameters.default.ratios
        elif isinstance(ratios, list):
            self.ratios  = np.array(ratios)
        if scales is None:
            self.scales  = utils_anchors.AnchorParameters.default.scales
        elif isinstance(scales, list):
            self.scales  = np.array(scales)

        self.num_anchors = len(ratios) * len(scales)
        self.anchors     = keras.backend.variable(utils_anchors.generate_anchors(
            base_size=size,
            ratios=ratios,
            scales=scales,
        ))

In keras_retinanet\bin\train.py 
    # load anchor parameters, or pass None (so that defaults will be used)
    anchor_params = None
    num_anchors   = None
    if config and 'anchor_parameters' in config:
        anchor_params = parse_anchor_parameters(config)
        num_anchors   = anchor_params.num_anchors()

    # Keras recommends initialising a multi-gpu model on the CPU to ease weight sharing, and to prevent OOM errors.
    # optionally wrap in a parallel model
    if multi_gpu > 1:
        from keras.utils import multi_gpu_model
        with tf.device('/cpu:0'):
            model = model_with_weights(backbone_retinanet(num_classes, num_anchors=num_anchors, modifier=modifier), weights=weights, skip_mismatch=True)
        training_model = multi_gpu_model(model, gpus=multi_gpu)
    else:
        model          = model_with_weights(backbone_retinanet(num_classes, num_anchors=num_anchors, modifier=modifier), weights=weights, skip_mismatch=True)
        #model = model_with_weights(backbone_retinanet(num_classes, backbone=backbone, modifier=modifier, anchor_parameters=anchor_parameters), weights=weights, skip_mismatch=True)
        training_model = model

In keras_retinanet\models\retinanet.py 
def retinanet(
    inputs,
    backbone_layers,
    num_classes,
    num_anchors             = None,
    create_pyramid_features = __create_pyramid_features,
    submodels               = None,
    name                    = 'retinanet'
):
    """ Construct a RetinaNet model on top of a backbone.

    This model is the minimum model necessary for training (with the unfortunate exception of anchors as output).

    Args
        inputs                  : keras.layers.Input (or list of) for the input to the model.
        num_classes             : Number of classes to classify.
        num_anchors             : Number of base anchors.
        create_pyramid_features : Functor for creating pyramid features given the features C3, C4, C5 from the backbone.
        submodels               : Submodels to run on each feature map (default is regression and classification submodels).
        name                    : Name of the model.

    Returns
        A keras.models.Model which takes an image as input and outputs generated anchors and the result from each submodel on every pyramid level.

        The order of the outputs is as defined in submodels:
        ```
        [
            regression, classification, other[0], other[1], ...
        ]
        ```
    """
    print("num_anchors: ", num_anchors)
    if num_anchors is None:
        num_anchors = AnchorParameters.default.num_anchors()

    if submodels is None:
        submodels = default_submodels(num_classes, num_anchors)

**Comment by hgaiser** [comment]
There is an entry for this in the FAQ, at the bottom of the README.md. It is not very verbose, but it should help get you started. Ideally you wouldn't need to change the code.

Regardless, this is not an issue with our code, so I'll close this issue. Please use the keras-retinanet Slack channel for questions on usage.

**Comment by reeldata** [comment]
I have actually encountered a similar error (shape mismatch at inference time) when passing my own anchor parameters to the retinanet-train script via the --config flag.

While trying to figure out what was going on with this I actually passed the default config.ini file (copy/pasted exactly from " keras-retinanet-test-data/config/config.ini") to retinanet-train and got the exact same error message as with the config file I created. As a work around, I went into "keras-retinanet/keras-retinanet/utils/anchors.py" and changed the default parameters (line 45) to the ones I wanted and it worked as intended.
